test/server/cards/02_SHD/units/ReyKeepingThePast.spec.ts

```diff
@@ -0,0 +1,93 @@
+describe('Rey, Keeping the Past', function() {
+    integration(function(contextRef) {
+        describe('Rey\'s Ability', function() {
+            it('ignores Heroism aspect penalty when unit Kylo unit is controlled', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['rey#keeping-the-past'],
+                        groundArena: ['kylo-ren#killing-the-past', 'pyke-sentinel'],
+                        base: 'dagobah-swamp',
+                        leader: 'darth-vader#dark-lord-of-the-sith'
+                    }
+                });
+
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.rey);
+
+                // Rey should cost 5 since it ignores the villainy aspect
+                expect(context.player1.countExhaustedResources()).toBe(5);
+            });
+
+            it('heals 2 and gives Shield token to a non-heroism unit', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['rey#keeping-the-past', 'pyke-sentinel', 'battlefield-marine']
+                    },
+                    player2: {
+                        groundArena: [{ card: 'wild-rancor', damage: 2 }, 'mandalorian-warrior']
+                    }
+                });
+
+                const { context } = contextRef;
+
+                const reset = (passAction = true) => {
+                    context.rey.exhausted = false;
+                    context.rey.damage = 0;
+                    context.wildRancor.damage = 2;
+                    context.pykeSentinel.damage = 0;
```

Comment:
Do we need to reset the state for Wild Rancor and Pyke Sentinel? Looks like they're each only involved in one test

Reply:
Just for the Wild Rancor as I use it for the attacks, removed the one for pyke sentinel.